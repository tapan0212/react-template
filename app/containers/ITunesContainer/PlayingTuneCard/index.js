/**
 *
 * PlayingTuneCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { colors } from '@themes';
import { Row, Col } from 'antd';
import { PauseCircleOutlined } from '@ant-design/icons';
import { FormattedMessage as T } from 'react-intl';

const CustomPauseButton = styled(PauseCircleOutlined)`
  float: right;
  font-size: 2rem;
  margin-top: 1rem;
  margin-right: 2rem;
  cursor: pointer;
`;
const CustomPlayingTuneCard = styled.div`
  position: fixed;
  bottom: 0;
  margin: 2vw;
  margin-left: 75vw;
  width: 20vw;
  height: 6rem;
  padding: 1rem;
  background-color: ${colors.primary};
  @media (max-width: 500px) {
    width: 60vw;
    margin-left: 35vw;
  }
`;
function PlayingTuneCard({ currentTune, dispatchCurrentTune }) {
  return (
    <CustomPlayingTuneCard data-testid="playing-tune-card">
      <Row>
        <Col span={12}>
          <div>
            <T id={currentTune.trackName} />
          </div>
          <div>
            <T id={currentTune.artistName} />
          </div>
        </Col>

        <Col span={12}>
          <CustomPauseButton
            data-testid="btn1"
            onClick={e => {
              dispatchCurrentTune(null);
            }}
          />
        </Col>
      </Row>
    </CustomPlayingTuneCard>
  );
}

PlayingTuneCard.propTypes = {
  currentTune: PropTypes.object,
  dispatchCurrentTune: PropTypes.func
};
export default PlayingTuneCard;
