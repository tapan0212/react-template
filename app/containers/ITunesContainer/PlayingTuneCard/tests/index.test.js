/**
 *
 * Tests for PlayingTuneCard
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import PlayingTuneCard from '../index';

describe('<PlayingTuneCard />', () => {
  let dispatchCurrentTuneSpy;
  const mockCurrentTune = {
    trackName: 'Perfect',
    artistName: 'Ed'
  };
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<PlayingTuneCard currentTune={mockCurrentTune} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 PlayingTuneCard component', () => {
    const { getAllByTestId } = renderWithIntl(<PlayingTuneCard currentTune={mockCurrentTune} />);
    expect(getAllByTestId('playing-tune-card').length).toBe(1);
  });

  it('should show stop button', () => {
    const { getByTestId } = renderWithIntl(
      <PlayingTuneCard currentTune={mockCurrentTune} dispatchCurrentTune={dispatchCurrentTuneSpy} />
    );
    expect(getByTestId('btn1')).toBeTruthy();
  });
});
