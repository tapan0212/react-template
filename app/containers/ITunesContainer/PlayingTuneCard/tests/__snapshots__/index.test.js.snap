// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayingTuneCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="PlayingTuneCard__CustomPlayingTuneCard-p3qjdi-1 dLfwxa"
      data-testid="playing-tune-card"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-12"
        >
          <div>
            Perfect
          </div>
          <div>
            Ed
          </div>
        </div>
        <div
          class="ant-col ant-col-12"
        >
          <span
            aria-label="pause-circle"
            class="anticon anticon-pause-circle PlayingTuneCard__CustomPauseButton-p3qjdi-0 fISzHk"
            data-testid="btn1"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="pause-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm-88-532h-48c-4.4 0-8 3.6-8 8v304c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8zm224 0h-48c-4.4 0-8 3.6-8 8v304c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
